### ISSUE TITLE
values for max_start_epoch

### ISSUE THREAD

**Issue by ankanbhunia** [GeorgeCazenavette/mtt-distillation#4]
Hi there, I can see that `max_start_epoch` is set to 20. However, during the generation of the expert trajectories, `train_epochs` is 50. It means that during distillation, we don't use most of the saved checkpoints (>20+3). My questions:
1. Is there any reason to choose `max_start_epoch ` as 20 not 50?
2. Can we make `train_epochs` to a lower value so to reduce training time?

**Comment by GeorgeCazenavette** [comment]
You're completely correct, feel free to lower the `train_epochs`.

We did use a later `max_start_epoch` for some experiments, but you can lower it for experiments that don't need that many.

![image](https://user-images.githubusercontent.com/18726777/160915720-7c8034b4-5675-498c-9985-309477612b8a.png)

Be aware that you'll actually need to train for `max_start_epoch` + `expert_epochs` epochs though.

**Comment by ankanbhunia** [comment]
Thanks @GeorgeCazenavette .

**Comment by ArmandXiao** [comment]
> You're completely correct, feel free to lower the `train_epochs`.
> 
> We did use a later `max_start_epoch` for some experiments, but you can lower it for experiments that don't need that many.
> 
> ![image](https://user-images.githubusercontent.com/18726777/160915720-7c8034b4-5675-498c-9985-309477612b8a.png)
> 
> Be aware that you'll actually need to train for `max_start_epoch` + `expert_epochs` epochs though.

**Hi, may I ask why `M` here is a lot less than `N`, which seems to be opposite of the saying shown in figure 3 that `N << M`.**
- ![image](https://user-images.githubusercontent.com/45375764/204597329-b86bfa7a-83d1-4f1f-89c3-416e6cfa92d5.png)

**Comment by GeorgeCazenavette** [comment]
That figure is referring to the total number of updates, showing that a few updates on synthetic data is equivalent to many updates on real data.

For simplicity, we just count the number of real epochs when setting that hyper-parameter.

**Comment by ArmandXiao** [comment]
> That figure is referring to the total number of updates, showing that a few updates on synthetic data is equivalent to many updates on real data.
> 
> For simplicity, we just count the number of real epochs when setting that hyper-parameter.

Thanks for the in time reply!

Just to clarify that we cannot find the corresponding `N` and `M` in the hyperparameter setting.
`M` and `N` are automatically adjusted by synthetic learning rate, and we have to find the number of iterations from actual runs.

**Comment by GeorgeCazenavette** [comment]
M and N always measure the same quantities.

The only difference is that in the figure, M is measured in _iterations_ to show that many real iterations is equivalent to few synthetic iterations.

Rather than save checkpoints at every iteration, we just save them at every epoch. So when setting --expert_epochs, that is M measured in epochs.

**Comment by ArmandXiao** [comment]
> M and N always measure the same quantities.
> 
> The only difference is that in the figure, M is measured in _iterations_ to show that many real iterations is equivalent to few synthetic iterations.
> 
> Rather than save checkpoints at every iteration, we just save them at every epoch. So when setting --expert_epochs, that is M measured in epochs.

I now understand. Thank you so much 😊

**Comment by zeyuanyin** [comment]
> 

Hi @GeorgeCazenavette. Thanks for your patience and careful reply. 

The table still confused me. `expert_epochs M+` only appears one time in the paper, and the value of `expert_update M` is not clear in the paper.

You point out `Be aware that you'll actually need to train for max_start_epoch + expert_epochs epochs though.`

Does it mean `expert_update M` = `synthetic_steps N` + `expert_epochs M+`?

For Tiny-ImageNet, I need to train each expert for at least `40 + 2 = 42` updates/epochs, right?
